File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.py
def solution(n):
    answer = 0
    
    for i in range(1, n + 1) :
        t = 0
        for j in range(i, n + 1) :
            t += j
            if t > n :
                break
            elif t == n :
                t = 0
                answer += 1
                break
    return answer

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        
        for (int i = 1; i <= n; i++) {
            if (is(i, n)) {
                answer++;
            }
        }
        return answer;
    }
    
    public boolean is(int i, int n) {
        int temp = 0;
        for (int j = i; i <= n; j++) {
            temp += j;
            if (temp == n) {
                return true;
            }
            if (temp > n) {
                return false;
            }
        }
        return false;
    }
}